python/dgl/core.py

```diff
@@ -228,13 +229,22 @@ def data_dict_to_list(graph, data_dict, func, target):
     if isinstance(func, fn.BinaryMessageFunction):
         if target in ["u", "v"]:
             output_list = [None] * graph._graph.number_of_ntypes()
-            for srctype, _, dsttype in graph.canonical_etypes:
+            # If there is only 1 source node, data_dict should be the value of feature,
+            # such as a tensor.
```

Comment:
`such as a tensor` or `namely, a tensor`? can it be other data types besides `tensor`?

Reply:
OK, I have changed it. I should be a tensor with 1 source node.